frontend-react/src/hooks/UseSessionStorage.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+
+import {
+    getStoredOrg,
+    getStoredSenderName,
+    setStoredOrg,
+    setStoredSenderName,
+} from "../contexts/SessionStorageTools";
+
+export interface SessionStore {
+    org?: string;
+    senderName?: string;
+}
+
+export interface SessionController {
+    values: SessionStore;
+    updateSessionStorage: (p: Partial<SessionStore>) => void;
+}
+
+const useSessionStorage = (): SessionController => {
+    const [values, setValues] = useState<SessionStore>({
+        org: getStoredOrg(),
+        senderName: getStoredSenderName(),
+    });
+
+    const updateSessionStorage = (newValues: Partial<SessionStore>) => {
+        setValues({
+            org: newValues.org,
+            senderName: newValues.senderName,
+        });
+    };
+
+    useEffect(() => {
+        if (values.org) {
```

Comment:
I'm confused. A) this will _only_ run when the value of `values.org` changes, and it's not setting `values.org`, it's setting the `sessionStorage` org to keep them congruent

Reply:
Yeah, you're right... there's no loop.

Maybe just check `if (values?.org?.length)` unless setting an empty string is a feature you want to allow.

Reply:
Just updated! :)